src/include/miopen/pooling/solvers.hpp

```diff
@@ -58,45 +58,47 @@ struct OldStyleSolver : SolverMixin<OldStyleProblemDescription>
         return IsApplicable(*std::get<0>(problem), *std::get<1>(problem));
     }
 
-    ConvSolution GetSolution(const OldStyleProblemDescription& problem) const
+    std::size_t GetWorkspaceSize(const OldStyleProblemDescription& problem) const override
     {
-        return GetSolution(*std::get<0>(problem), *std::get<1>(problem));
+        return GetWorkspaceSize(*std::get<0>(problem), *std::get<1>(problem));
     }
 
-    std::size_t GetWorkspaceSize(const OldStyleProblemDescription& problem) const override
+    ConvSolution GetSolution(const OldStyleProblemDescription& problem) const
     {
-        return GetWorkspaceSize(*std::get<0>(problem), *std::get<1>(problem));
+        return GetSolution(*std::get<0>(problem), *std::get<1>(problem));
     }
 
     virtual bool IsApplicable(const ExecutionContext& context,
-                              const miopen::pooling::ProblemDescription& problem) const        = 0;
-    virtual ConvSolution GetSolution(const ExecutionContext& context,
-                                     const miopen::pooling::ProblemDescription& problem) const = 0;
+                              const miopen::pooling::ProblemDescription& problem) const = 0;
     virtual std::size_t
     GetWorkspaceSize(const ExecutionContext& context,
-                     const miopen::pooling::ProblemDescription& problem) const = 0;
+                     const miopen::pooling::ProblemDescription& problem) const                 = 0;
+    virtual ConvSolution GetSolution(const ExecutionContext& context,
```

Comment:
This change is simply a move of existing code. If you would like to re-organize the code my moving its parts, then I highly recommend doing that in a dedicated NFC PR. Why: if this PR is reverted due to some reason, then the code move changes will be reverted too, and that would increase the probability of merge conflicts.

(Applies to many changes in this PR)

Reply:
Changes have been reverted. Thanks!